internal/grpc/services/storageprovider/storageprovider.go

```diff
@@ -491,14 +500,42 @@ func (s *service) DeleteStorageSpace(ctx context.Context, req *provider.DeleteSt
 }
 
 func (s *service) CreateContainer(ctx context.Context, req *provider.CreateContainerRequest) (*provider.CreateContainerResponse, error) {
-	newRef, err := s.unwrap(ctx, req.Ref)
+	var err error
+	var parentRef *provider.Reference
+	var name string
+	switch {
+	case utils.IsRelativeReference(req.Ref):
+		req.Ref.Path, name = path.Split(req.Ref.Path)
+		parentRef = req.Ref
+	case utils.IsAbsoluteReference(req.Ref):
+		ref, err := s.unwrap(ctx, req.Ref)
+		if err != nil {
+			return &provider.CreateContainerResponse{
+				Status: status.NewInternal(ctx, err, "error unwrapping path"),
+			}, nil
+		}
+		parentRef = &provider.Reference{Path: path.Dir(ref.GetPath())}
+		name = path.Base(ref.GetPath())
+	default:
+		return &provider.CreateContainerResponse{
+			Status: status.NewInvalidArg(ctx, "invalid reference, name required"),
+		}, nil
+	}
+	var st *rpc.Status
 	if err != nil {
```

Comment:
`err` will always be nil here. We already check the error returned by `unwrap` in the switch case and return, and it is assigned to a local var anyway; so we can remove this block